### PR TITLE
manila: Add pagination in list API

### PR DESCRIPTION
**Pull request by kpawar-sap**
Fixes https://github.com/sapcc/openstack-nannies/issues/64

### CODE REVIEW

File: scripts/manilananny.py [hunk 1]
```diff
@@ -119,7 +119,14 @@ def list_shares(self, status=None):
             search_opts = {'all_tenants': 1}
             if status:
                 search_opts.update({'status': status})
-            return self.manilaclient.shares.list(search_opts=search_opts)
+
+            shares = []
+            offset = 0
+            for i in range(10):
```
Comment (Carthaca) [inline scripts/manilananny.py:125]: Please use the `with_count` parameter to determine the total number of shares - we have already more than 10k shares in our big regions.

**Review by Carthaca — CHANGES_REQUESTED** [review]
Similar to the inline comment regarding total number of shares, we may get more than 5k snapshots one day, please consider this

File: scripts/manilananny.py (hunk 1 above)
Comment (kpawar-sap) [reply scripts/manilananny.py]: done

File: scripts/manilananny.py [hunk 2]
```diff
@@ -119,7 +120,19 @@ def list_shares(self, status=None):
...
-            return self.manilaclient.shares.list(search_opts=search_opts)
+
+            opts = copy.deepcopy(search_opts)
+            opts.update({'with_count': True})
+            _, count = self.manilaclient.shares.list(search_opts=opts)
+            count = (count / 1000) + 1
+
+            shares = []
+            offset = 0
+            for i in range(count):
```
Comment (Carthaca) [inline scripts/manilananny.py:131]: This needs an integer conversion

```suggestion
            for i in range(int(count)):
```

**Review by Carthaca — CHANGES_REQUESTED** [review]

File: 1
Comment (Carthaca) [inline 1:1]: Has that file been committed by accident?

**Review by Carthaca — CHANGES_REQUESTED** [review]

File: 1
Comment (kpawar-sap) [reply 1:1]: yes, removed

File: nannies/manila-missing-snapshot.py [hunk 3]
```diff
@@ -81,7 +81,18 @@ def _get_missing_snapshots(self):
 
         logging.info("fetching Manila Snapshots...")
 
-        for _snapshot in manila.share_snapshots.list(search_opts={'all_tenants': True}):
+        snapshots = []
+        offset = 0
+        for i in range(100):
+            snaps = manila.share_snapshots.list(
+                search_opts={'all_tenants': True, 'limit': 1000, 'offset': offset})
```
Comment (Carthaca) [inline nannies/manila-missing-snapshot.py:88]: Now that we have the `with_count` option, can you please add it here, too?

**Review by Carthaca — CHANGES_REQUESTED** [review]
Can you please pick this up?

File: scripts/manilananny.py [hunk 4]
```diff
@@ -148,7 +161,23 @@ def list_share_snapshots(self, status=None):
             search_opts = {'all_tenants': 1}
             if status:
                 search_opts.update({'status': status})
-            return self.manilaclient.share_snapshots.list(search_opts=search_opts)
+
+            opts = copy.deepcopy(search_opts)
+            opts.update({'with_count': True})
```
Comment (chuan137) [inline scripts/manilananny.py:166]: is `with_count` an option for `snapshot-list` too?

Comment (chuan137) [reply scripts/manilananny.py:166]: I see, manilaclient needs an update

**Review by chuan137 — APPROVED** [review]
lgtm

**Pull request merged**